src/passes/OptimizeInstructions.cpp

```diff
@@ -1663,6 +1663,16 @@ struct OptimizeInstructions
         }
       }
     }
+    {
+      // x * 2.0  ==>  x + x
+      Expression* x;
+      if (matches(curr, binary(Mul, any(&x), fval(2.0))) &&
+          (x->is<LocalGet>() || x->is<GlobalGet>())) {
+        curr->op = Abstract::getBinary(type, Abstract::Add);
+        curr->right = ExpressionManipulator::copy(x, *getModule());
```

Comment:
@tlively is it necessary expression copy here or simple assignment `curr->right = x` may also safely works?

Reply:
Unfortunately the copy is necessary.